src/ansys/aedt/core/visualization/advanced/rcs_visualization.py

```diff
@@ -0,0 +1,1991 @@
+from ansys.aedt.core.generic.constants import unit_converter
+from ansys.aedt.core.generic.general_methods import conversion_function
+from ansys.aedt.core.generic.general_methods import open_file
+from ansys.aedt.core.generic.general_methods import pyaedt_function_handler
+from ansys.aedt.core.visualization.plot.matplotlib import ReportPlotter
+from ansys.aedt.core.visualization.plot.pyvista import ModelPlotter
+from ansys.tools.visualization_interface import MeshObjectPlot
+from ansys.tools.visualization_interface import Plotter
+from ansys.tools.visualization_interface.backends.pyvista import PyVistaBackend
+import numpy as np
+import pandas as pd
+import pyvista as pv
+import scipy.interpolate
+
+
+class MonostaticRCSData(object):
+    """Provides monostatic radar cross-section (RCS) data.
+
+    Read monostatic RCS metadata in a JSON file generated by :func:`MonostaticRCSExporter` and return the
+    Python interface to plot and analyze the RCS data.
+
+    Parameters
+    ----------
+    input_file : str
+        Metadata information in a JSON file.
+
+    Examples
+    --------
+    >>> from ansys.aedt.core import Hfss
+    >>> from ansys.aedt.core.generic.rcs_visualization import MonostaticRCSData
+    >>> app = Hfss(version="2023.2", design="Antenna")
+    >>> data = app.get_rcs_data()
+    >>> metadata_file = data.metadata_file
+    >>> app.release_desktop()
+    >>> rcs_data = MonostaticRCSData(input_file=metadata_file)
+    """
+
+    def __init__(self, input_file):
+
+        # Public
+        self.output_dir = os.path.dirname(input_file)
+
+        if not os.path.isfile(input_file):  # pragma: no cover
+            raise Exception("JSON file does not exist.")
+
+        # Private
+        self.__logger = logger
+        self.__input_file = input_file
+        self.__raw_data = {}
+        self.__frequency = None
+        self.__freq_index = 0
+        self.__model_units = "meter"
+        self.__name = None
+        self.__solution = None
+
+        self.__incident_wave_theta = None
+        self.__incident_wave_phi = None
+        self.__available_incident_wave_theta = None
+        self.__available_incident_wave_phi = None
+
+        self.__frequencies = []
+
+        self.__model_info = {}
+        self.__component_objects = None
+
+        with open_file(input_file) as f:
+            self.__metadata = json.load(f)
+
+        if not self.metadata:  # pragma: no cover
+            raise Exception("Metadata could not be loaded..")
+
+        self.__frequency_units = self.metadata["frequency_units"]
+
+        self.__monostatic_file = os.path.join(self.output_dir, self.metadata["monostatic_file"])
+
+        self.__data_conversion_function = "dB20"
+        self.__window = "Flat"
+        self.__window_size = 1024
+        self.__aspect_range = "Horizontal"
+        self.__upsample_range = 512
+        self.__upsample_azimuth = 64
+
+        if not os.path.isfile(self.__monostatic_file):  # pragma: no cover
```

Comment:
Could you add some unit test to cover cases with `#pragma:  no cover` ? I feel like this can be easily performed with this class